quickevent/app/quickevent/plugins/CardReader/src/cardreaderplugin.cpp

```diff
@@ -470,8 +470,9 @@ bool CardReaderPlugin::processCardToRunAssignment(int card_id, int run_id)
 		if(start_time.isNull() && leg > 1) {
 			/// if start time not set, take start time from previous leg
 			q.execThrow("SELECT finishTimeMs FROM runs"
-					 " WHERE relayId=" + QString::number(relay_id)
-				   + " AND leg=" + QString::number(leg-1));
+					  " WHERE relayId=" + QString::number(relay_id)
+					+ " AND leg=" + QString::number(leg-1)
+					+ " ORDER BY finishTimeMs");
 			if(q.next()) {
```

Comment:
Does the query above always return only one result? Replace this `if` with `while`?

Reply:
it can return multiple results, however, i am interested only in the faster one. so i order by finish time and read only the first row using `if`